core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -812,4 +815,32 @@ private boolean deleteProperties(Namespace namespace, Set<String> properties) {
   protected Map<String, String> properties() {
     return catalogProperties == null ? ImmutableMap.of() : catalogProperties;
   }
+
+  @Override
+  public TableBuilder buildTable(TableIdentifier identifier, Schema schema) {
+    return new ViewAwareTableBuilder(identifier, schema);
+  }
+
+  /**
+   * The purpose of this class is to add view detection only when SchemaVersion.V1 schema is used
+   * when replacing a table.
+   */
+  protected class ViewAwareTableBuilder extends BaseMetastoreCatalogTableBuilder {
+
+    private final TableIdentifier identifier;
+
+    public ViewAwareTableBuilder(TableIdentifier identifier, Schema schema) {
+      super(identifier, schema);
+      this.identifier = identifier;
+    }
+
+    @Override
+    public Transaction replaceTransaction() {
+      if (schemaVersion == JdbcUtil.SchemaVersion.V1 && viewExists(identifier)) {
```

Comment:
@nastra, couldn't we just throw `UnsupportedOperationException` in `viewExists` with schema v0?

Reply:
That was the initial behavior, but it breaks some behavior when using `V0` schema.

Reply:
Ah, thanks. You're right. I was thinking the REST catalog change would catch the exception, but that is a separate builder. Looks good to me.